aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -131,74 +181,35 @@ pub(crate) fn run_script_prologue(
                     .as_ref()
                     .map(|proof| proof.optional_auth_key()),
             ) {
-                let serialized_args = vec![
-                    serialized_signers.sender(),
+                serialized_args.insert(
```

Comment:
Inserting at indices with magic numbers seems very error-prone to me, can we do
```rust
let mut serialized_args = vec![
            serialized_signers.sender(),
];
if .. {
  // push fee payer
}
// push txn_authentication_key
if .. {
  // push fee_payer_auth_key
}
// the rest can be a .extend(!vec[ /* remaining args here */])
```

Reply:
It was getting a little confusing with all the if else statements. I moved back to how it was originally there before this PR.